### PR TITLE
[wip] IF textual inversion

### PR DESCRIPTION
**Pull request by williamberman**

### CODE REVIEW

File: examples/textual_inversion/textual_inversion.py [hunk 1]
```diff
@@ -825,36 +917,81 @@ def main():
...
+
+                    # simple loss - predicting noise, x0, or x_prev
+
+                    pred_noise, _ = model_pred.chunk(2, dim=1)
+
+                    simple_losses = F.mse_loss(pred_noise, noise)
+
+                    vb_loss_weight = noise_scheduler.config.num_train_timesteps / 1000.0
+
+                    loss = simple_losses + vb_losses.mean() * vb_loss_weight
```
Comment (patrickvonplaten) [inline examples/textual_inversion/textual_inversion.py:984]: If we don't consider the variance we just have normal MSE loss - think that's something worth trying out to begin with. Could we run a first experiment just setting `vb_loss_weight=0` ?

**Comment by williamberman** [comment]
closing this for now, training is potentially non-trivial

File: examples/textual_inversion/textual_inversion.py [hunk 2]
```diff
@@ -825,36 +922,85 @@ def main():
...
+                    model_mean, _, model_log_variance, _ = noise_scheduler.p_mean_variance(
+                        x=noisy_input_images, t=timesteps, model_output=model_pred
+                    )
 
-                # Get the target for loss depending on the prediction type
-                if noise_scheduler.config.prediction_type == "epsilon":
-                    target = noise
-                elif noise_scheduler.config.prediction_type == "v_prediction":
-                    target = noise_scheduler.get_velocity(latents, noise, timesteps)
+                    # kl loss with detached model predicted mean, for stability reasons as in paper
```
Comment (sayakpaul) [inline examples/textual_inversion/textual_inversion.py:966]: DDPM paper?

File: examples/textual_inversion/textual_inversion.py [hunk 3]
```diff
@@ -825,36 +922,85 @@ def main():
...
+                    pred_noise, _ = model_pred.chunk(2, dim=1)
+
+                    simple_losses = F.mse_loss(pred_noise, noise)
+
+                    vb_loss_weight = noise_scheduler.config.num_train_timesteps / 1000.0
+
+                    loss = simple_losses + vb_losses.mean() * vb_loss_weight
                 else:
-                    raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")
+                    if model_pred.shape[1] == 6:
```
Comment (sayakpaul) [inline examples/textual_inversion/textual_inversion.py:992]: What's this for?
